src/io/builder_proxy.rs

```diff
@@ -33,10 +36,20 @@ impl ColumnBuilderProxy for StringColumnBuilderProxy {
     generic_trait_impl!(VecT::U64);
     fn add(&mut self, string: &str, dictionary: Option<&mut Dict>) -> Result<(), Error> {
         self.commit();
+
+        // TODO: parsing of DSV and Program should be unified
+        // TODO: DSV parsing should depend on logical types
+        let parsed = parse_dsv_constant(string)
+            .map(|(remainder, parsed)| {
+                debug_assert!(remainder.is_empty()); // the whole value should be parsed
+                parsed
+            })
+            .unwrap_or(string.to_string());
```

Comment:
```suggestion
        let parsed = super::parser::all_input_consumed(parse_dsv_constant)(string)
            .unwrap_or(string.to_string());
```

Checked the performance and it has been about 3 seconds faster on snomed to use `all_input_consumed` instead of the `map`-approach